src/module.cc

```diff
@@ -145,22 +144,63 @@ static void OnFatalError(const char* location, const char* message) {
     TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, nullptr);
   }
   fflush(stderr);
-  if (nodereport_core) {
-    raise(SIGABRT); // core dump requested (default)
-  } else {
-    exit(1); // user specified that no core dump is wanted, just exit
-  }
+  raise(SIGABRT);
 }
 
 bool OnUncaughtException(v8::Isolate* isolate) {
-   // Trigger NodeReport if required
+  // Trigger NodeReport if requested
   if (nodereport_events & NR_EXCEPTION) {
     TriggerNodeReport(isolate, kException, "exception", __func__, nullptr);
-  } 
-  return nodereport_core;
+  }
+  if ((commandline_string.find("abort-on-uncaught-exception") != std::string::npos) ||
+      (commandline_string.find("abort_on_uncaught_exception") != std::string::npos)) {
+    return true;  // abort required
+  }
+  // On Node v4 and v6 we need to print the exception backtrace to stderr, as
+  // Node does not do so (unless we trigger an abort, as above)
+  if ((version_string.find("v4") != std::string::npos) ||
+      (version_string.find("v6") != std::string::npos)) {
+    fprintf(stderr, "\nUncaught exception at:\n");
+#ifdef _WIN32
+    // On Windows, print the stack using StackTrace API
+    PrintStackFromStackTrace(isolate, stderr);
+#else
+    // On other platforms use the Message API
+    Message::PrintCurrentStackTrace(isolate, stderr);
+#endif
+  }
+  return false;
 }
 
-#ifndef _WIN32
+#ifdef _WIN32
+static void PrintStackFromStackTrace(Isolate* isolate, FILE* fp) {
```

Comment:
It's a simplified version of the function in node-report.cc. The stack trace in a NodeReport has quite a bit more detail, e.g. the pc value on each line and the JavaScript state value. I did consider sharing the code but I think the resulting single function would have been more complicated. The order of the parameters matches Message::PrintCurrentStackTrace() which is called on the next line in OnUncaughtException()